src/Api/HL/APIException.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Api\HL;
+
+use Throwable;
+
+/**
+ * An exception thrown by the API.
+ * A user message can be provided to be displayed to the user.
+ * Otherwise, only a generic message will be displayed to the user.
+ */
+class APIException extends \Exception
+{
+    private string $user_message;
+
+    public function __construct(string $message = '', string $user_message = '', int $code = 0, ?Throwable $previous = null)
+    {
+        if ($user_message === '') {
+            $user_message = __('An error occurred while processing your request.');
+        }
+        if ($message === '') {
+            $message = $user_message;
```

Comment:
IMHO, we should ensure to always have a non translated message that goes into logs. Indeed, it would be a annoying to have to analyze an error trace that contains a message translated into an alphabet we do not even know.